river/Cursor.zig

```diff
@@ -744,25 +856,29 @@ pub fn enterMode(self: *Self, mode: enum { move, resize }, view: *View) void {
     self.setImage(if (mode == .move) .move else .@"se-resize");
 }
 
-/// Return from down/move/resize to passthrough
-fn leaveMode(self: *Self, event: *wlr.Pointer.event.Button) void {
+/// Return from disabled/down/move/resize to passthrough
+pub fn leaveMode(self: *Self, event: ?*wlr.Pointer.event.Button) void {
     log.debug("leave {s} mode", .{@tagName(self.mode)});
 
     switch (self.mode) {
         .passthrough => unreachable,
+        .disabled => {}, // TODO: what now? reset the image? prolly that + restore ptr capability on seat
         .down => {
             // If we were in down mode, we need pass along the release event
-            _ = self.seat.wlr_seat.pointerNotifyButton(event.time_msec, event.button, event.state);
+            _ = self.seat.wlr_seat.pointerNotifyButton(event.?.time_msec, event.?.button, event.?.state);
         },
         .move => {},
         .resize => |resize| resize.view.setResizing(false),
     }
 
     self.mode = .passthrough;
-    self.passthrough(event.time_msec);
+    // this is only a temporary solution. a fake event has to be created if
+    // there is none (TODO)
+    if (event != null) self.passthrough(event.?.time_msec);
```

Comment:
This seems to be a problem when showing the cursor manually (via riverctl, see `command/config.zig`). Do you have an idea how to solve this?

Now thinking about it, *maybe* the issue could be that i don’t refocus the view under cursor though...

Reply:
Sorry for rubber-ducking, the main issue i had seems to be solved.

But still i’m not sure what to do if there’s no event to pass to `leaveMode`. Make it up?